pkg/cri/server/sandbox_stats_list.go

```diff
@@ -34,6 +36,10 @@ func (c *criService) ListPodSandboxStats(
 	podSandboxStats := new(runtime.ListPodSandboxStatsResponse)
 	for _, sandbox := range sandboxes {
 		sandboxStats, err := c.podSandboxStats(ctx, sandbox)
+		if errdefs.IsUnavailable(err) {
```

Comment:
switch maybe and have a look at the rest of the errors that might happen.. looking to be convinced that default should be **do not return any stats if there is one pod in the list with an error outside unavailable** 

might be better to return the stats we do have and the last error or a list of errors..

Reply:
(James is OOO and I'm helping to push this along) it looks like the only errors returned by `c.podSandboxStats` are `IsUnavialble` or unspecified errors. Because of this is it OK to keep this as if statements instead of using a switch?

also - I make some updates to aggregate errors and also return stats for all sandboxes we are able to get stats for if there are any errors - PTAL

Reply:
Also i see above that most of the error should be treated as transient.

> I believe most of the time an error is treated as transient. if it's worth logging in the kubelet, then I'd report it. If it's part of normal state changes in containerd, then I would say hide the errors. e.g: is there action a user can take, or a maintainer should be aware of? if so, error. if not, hide

Reply:
it's ok but not preferred .. the switch comment is a standard nit/style preference normally taken by our stack components for using switch vs a sequence of if then else blocks..  normally because sequences of if then else blocks can often hide technical issues/bugs 

> Also i see above that most of the error should be treated as transient.

That's what I figured... thanks for looking into it.

Reply:
> it's ok but not preferred .. the switch comment is a standard nit/style preference normally taken by our stack components for using switch vs a sequence of if then else blocks.. normally because sequences of if then else blocks can often hide technical issues/bugs
> 
> > Also i see above that most of the error should be treated as transient.
> 
> That's what I figured... thanks for looking into it

I'll make an update to use a switch statement